src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java

```diff
@@ -179,10 +180,28 @@ public void onAnvilPrepare(@NotNull final PrepareAnvilEvent event) {
 					&& player.hasPermission("minetinker.tool.combine")) {
 				newTool = item1.clone();
 
-				for (Modifier tool2Mod : modManager.getToolMods(item2)) {
-					int modLevel = modManager.getModLevel(item2, tool2Mod);
+				final List<Modifier> newToolMods = modManager.getToolMods(newTool);
+				final List<Modifier> item2Mods = modManager.getToolMods(item2);
+				final List<Modifier> sharedModifiers = item2Mods.stream().filter(newToolMods::contains).toList();
+				final List<Modifier> uniqueModifiers = item2Mods.stream().filter(Predicate.not(newToolMods::contains)).toList();
+
+				for (var shared : sharedModifiers) {
+					final int newToolModLevel = modManager.getModLevel(newTool, shared);
+					final int item2ModLevel = modManager.getModLevel(item2, shared);
+
+					if (item2ModLevel > newToolModLevel) { //If the 2nd tool's modifier has a higher level, clamp the level to the 2nd tool's level.
```

Comment:
Is it necessary to do it the other way around as well?

Reply:
On further thinking, it is not necessary as no changes need to be made to the modifier on the tool, but maybe an explaining comment would help as it is not in plain site.

Reply:
@X1nto

Reply:
Basically, if the item2's enchant level is higher than the item1's enchant level, the outcome level is the one from item2. if the levels are the same, the outcome is the item2 level + 1, and if item2's level is lower than item1's, the outcome is item1's level.

Reply:
By default, the outcome is item1's level with checks for the remaining 2 situations added later, so the check for when item2 < item1 is not necessary